app/src/main/kotlin/com/wire/android/ui/userprofile/other/bottomsheet/OtherUserBottomSheetState.kt

```diff
@@ -21,21 +25,29 @@ class OtherUserBottomSheetState {
     }
 
     private fun updateBottomSheetContentIfNeeded() {
-        when (bottomSheetContentState) {
-            is BottomSheetContent.Conversation -> toConversation()
-            is BottomSheetContent.ChangeRole -> toChangeRole()
-            null -> {}
+        when (flag) {
```

Comment:
Open question, I was looking at options here, and I was thinking if we really need this `flag` field 🤔 
Is it possible to achieve the same with `conversationSheetContent.bottomSheetContentState` field and check for the types ?
- instance of `BottomSheetContent.Conversation`
- instance of `BottomSheetContent. ChangeRole `
- and null

Reply:
it was like that before it, but it doesn't work anymore. 

As far as we load the content only on user clicks and not on opening the screen, the `bottomSheetContentState` is null when user just clicks. Then `conversationSheetContent` is loaded and calls `updateBottomSheetContentIfNeeded()` but `bottomSheetContentState` is still null (not instance of `BottomSheetContent.Conversation`) so nothing happens.

Reply:
Oh, I see! Thanks 😅